demo/AutoLoader.php

```diff
@@ -0,0 +1,188 @@
+<?php
```

Comment:
So the answer is no. Isn't one simple function enough to implement PSR-4 autoloader? The autoloader you've added in this PR is huge.